fix SecurityRequirementsOperationFilter in case of minimal API

hartmair opened the pull request:
What's in this PR:
- [x] Fixes SecurityRequirementsOperationFilter in case of minimal API where both `MethodInfo` and `ApiDescription.ActionDescriptor.EndpointMetadata` are provided introduced in https://github.com/domaindrivendev/Swashbuckle.AspNetCore/pull/2356, i.e. Swashbuckle.AspNetCore.SwaggerGen Version 6.3.0 or later
Fixes #227 
- [ ] Bump Swashbuckle.AspNetCore.* dependencies to latest version as this breaks other filters because of Newtonsoft changes
- [x] Bump Swashbuckle.AspNetCore.* dependencies in MinimalApi6.0-Swashbuckle6 project to latest version
- [x] .NET 6
- [ ] .NET 7
- [ ] Remove deprecated .NET 5[^1]
- [ ] Remove deprecated .NET 3.1[^1]
- [ ] Remove deprecated .NET 2.1[^1]
- [x] Bump CI pipeline to VS 2022
- [x] Integration Tests
- [ ] Proper unit tests for #225 

[^1]: https://dotnet.microsoft.com/en-us/platform/support/policy/dotnet-core

andrew-yustyk commented:
As you have a big refactoring for the OperationFilterContextExtensions, maybe you can fix issue #225 (PR: #226 ) ?
There's a need just to change `DeclaringType` to `ReflectedType`

hartmair commented:
@mattfrear PR is ready

mattfrear commented:
I just tried to run your feature branch with my test/WebApi3.1 test project, and it throws an exception:
```csharp
System.ArgumentException
  HResult=0x80070057
  Message=An item with the same key has already been added. Key: CustomHeader
  Source=System.Private.CoreLib
  StackTrace:
   at System.ThrowHelper.ThrowAddingDuplicateWithKeyArgumentException[T](T key)
   at System.Collections.Generic.Dictionary`2.TryInsert(TKey key, TValue value, InsertionBehavior behavior)
   at System.Collections.Generic.Dictionary`2.Add(TKey key, TValue value)
   at Swashbuckle.AspNetCore.Filters.AddResponseHeadersFilter.Apply(OpenApiOperation operation, OperationFilterContext context) in C:\code\github\Swashbuckle.AspNetCore.Filters\src\Swashbuckle.AspNetCore.Filters\ResponseHeaders\AddResponseHeadersFilter.cs:line 29
   at Swashbuckle.AspNetCore.SwaggerGen.SwaggerGenerator.GenerateOperation(ApiDescription apiDescription, SchemaRepository schemaRepository)
   at Swashbuckle.AspNetCore.SwaggerGen.SwaggerGenerator.GenerateOperations(IEnumerable`1 apiDescriptions, SchemaRepository schemaRepository)
   at Swashbuckle.AspNetCore.SwaggerGen.SwaggerGenerator.GeneratePaths(IEnumerable`1 apiDescriptions, SchemaRepository schemaRepository)
   at Swashbuckle.AspNetCore.SwaggerGen.SwaggerGenerator.GetSwagger(String documentName, String host, String basePath)
   at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.<Invoke>d__4.MoveNext()
```

![image](https://github.com/mattfrear/Swashbuckle.AspNetCore.Filters/assets/169179/405bf0ed-bf0b-4217-9108-8dee0283c685)

hartmair commented:
@mattfrear 

> I just tried to run your feature branch with my test/WebApi3.1 test project, and it throws an exception:

I have added integration tests and updated the implementation which led to much better solution :)

mattfrear commented:
Just a note to say I haven't forgotten about this. 

I am on holiday with my family, and I was hoping I would have a chance to look at it during my holiday, but that is unrealistic. So I won't be able to look at this until early September.

Sorry about the delay.

mattfrear commented on test/MinimalApi6.0-Swashbuckle6/Program.cs:
```diff
@@ -1,9 +1,13 @@
 
+
 // Add services to the container.
+builder.Services.AddControllers();
```
As per its name, this test project is just for testing Minimal API and not controllers, so it shouldn't have this.

mattfrear commented:
Thanks again for this - as discussed here https://github.com/mattfrear/Swashbuckle.AspNetCore.Filters/issues/227#issuecomment-1707502143
